src/Controller/Entry/EntryFrontController.php

```diff
@@ -5,213 +5,132 @@
-            ->setTime($criteria->resolveTime($time))
-            ->setType($criteria->resolveType($type));
-        $criteria->moderated = true;
-
-        // We do not set language filter for moderated view.
+            $content_tmpl = 'entry/';
+            $content_key = 'entries';
+        } elseif ('microblog' === $content) {
+            $posts = $this->postRepository->findByCriteria($criteria);
 
-        $method = $criteria->resolveSort($sortBy);
-        $listing = $this->$method($criteria);
+            $content_tmpl = 'post/';
+            $content_key = 'posts';
+        }
 
         if ($request->isXmlHttpRequest()) {
             return new JsonResponse(
                 [
                     'html' => $this->renderView(
-                        'entry/_list.html.twig',
+                        $content_tmpl.'_list.html.twig',
                         [
-                            'entries' => $listing,
+                            $content_key => $posts,
                         ]
                     ),
                 ]
             );
         }
 
         return $this->render(
-            'entry/front.html.twig',
+            $content_tmpl.'front.html.twig',
             [
-                'entries' => $listing,
+                $content_key => $posts,
+                'criteria' => $criteria,
             ]
         );
     }
 
-    #[IsGranted('ROLE_USER')]
-    public function favourite(?string $sortBy, ?string $time, ?string $type, Request $request): Response
+    // $name is magazine name, for compatibility
+    public function front_redirect(?string $sortBy, ?string $time, ?string $type, string $federation, string $content, ?string $name, Request $request): Response
     {
-        $criteria = new EntryPageView($this->getPageNb($request));
-        $criteria->showSortOption($criteria->resolveSort($sortBy))
-            ->setFederation(
-                'false' === $request->cookies->get(
-                    ThemeSettingsController::KBIN_FEDERATION_ENABLED,
-                    true
-                ) ? Criteria::AP_LOCAL : Criteria::AP_ALL
-            )
-            ->setTime($criteria->resolveTime($time))
-            ->setType($criteria->resolveType($type));
-        $criteria->favourite = true;
-
-        // No language criteria for favourites, either
-
-        $method = $criteria->resolveSort($sortBy);
-        $listing = $this->$method($criteria);
-
-        if ($request->isXmlHttpRequest()) {
-            return new JsonResponse(
-                [
-                    'html' => $this->renderView(
-                        'entry/_list.html.twig',
-                        [
-                            'entries' => $listing,
-                        ]
-                    ),
-                ]
-            );
+        $user = $this->getUser(); // Fetch the user
+        $subscription = $this->subscriptionFor($user); // Determine the subscription filter based on the user
+
+        if ($magazine) {
```

Comment:
`$magazine` is never defined. It is not available in this scope

Reply:
Ah, good spot, that's embarrassing. And... it's easy to fix the typo $magazine -> $name, but it led me to a whole area of microblog routing that seems like will need some careful attention to make sure is working properly under the new scheme. I'll look into it + make sure it's right but it'll take a little time.

Reply:
Okay, I think I sorted it out; I *think* it's enough to just add the right parameters to the magazine_posts redirect and everything will work. I tested it lightly but not real intensively.